pkg/operation/botanist/component/istio/cmputils.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istio
```

Comment:
```suggestion
package istio_test
```
Since these methods appear to only be used in the `istio_test` package and so that they do not get imported every time the `istio` package is inported

Reply:
Actually, they are not used in `istio_test` at all, but in `kubeapiserverexposure_test`, `vpnauthzserver_test` and `vpnseedserver_test`. This is why I added them simply to `istio` package to have them in a common place where they fit.
Feel free to suggest a different place, but `istio_test` does not work as the functions need to be imported in three different locations.

Reply:
Ah sorry. You could move the `BeComparableToMatcher` matcher to `pkg/utils/test/matchers`. 
Not sure what would be the best place for the `CmpOptsFor...` helper functions though.

Reply:
Any reason why `CmpOptsFor...` should not be in `pkg/utils/test/matchers` as well? Those functions only make sense for the `BeComparableToMatcher` (or if `go-cmp` would be used separately).

Reply:
My reasoning would be that `BeComparableToMatcher` is pretty generic and can be used for other tests as well, whereas the `CmpOptsFor...` are particularly related to istio. 
I dug a bit in our folders and I think you could place the helper functions in e.g. `pkg/botanist/component/test/istiocomponent.go`

Reply:
Split the content into `pkg/utils/test/matchers` and `pkg/botanist/component/test/istiocomponent.go` as suggested.
Let me know in case further changes are required.